field/TinyLang/Field/Generator.hs

```diff
@@ -368,19 +310,27 @@ even though shrinking to the body of the let-expression would also be correct, b
 shrinking is type-preserving, we let the type-preserving shrinker do it.
 -}
 
+-- We can shrink any expression to just a hardcoded ground value (except we shouldn't shrink other
+-- ground values to hardcoded ground values to prevent looping).
+defaultUniVal :: forall f a. (KnownUni f a, Field f) => UniVal f a
+defaultUniVal = case knownUni @f @a of
+    Bool  -> UniVal Bool True
+    Field -> UniVal Field $ fromInteger 100
```

Comment:
If we're working over a field of characteristic 2 or 5 this will be zero, which may not be what we want.  Unfortunately there isn't anything that will be nonzero for all possible characteristics (because every integer is divisible by at least one prime p, nad then it'll map onto 0 in Z_p).

Reply:
Oh, hold on: 1 will always be nonzero, but maybe it's not very arbitrary.

Reply:
We can have a type class `HasSomeBigNumber`, but it's kind of weird. We can also run `unsafePerformIO $ generate arbitrary`, but I don't really want to use `unsafePerformIO` without a good justification. Good point, though, I'll change the number to 101, which is at least prime.

There is another weird option: we can add annotations to the AST (like we do in Plutus) and put there arbitrary ground values (unrelated to the rest of the AST) and then shrink to those. It might be not that bad, I'm not sure. What do you think?

Reply:
Changed the number to 101 for the time being.

Reply:
Maybe the `Field` class should have something that tells you what the characteristic is.  Sometimes it's important to know that: for example, there are plenty of theorems that are true for all fields except those of characteristic two (eg, maybe you have to calculate (a+b)/2 at some point, but you can't divide by by 2 because it's that same as 0).

Reply:
And if characteristic is 0, then we pick `100`? If we're going to shrink to the biggest possible number (which is what characteristic is, for a finite field, right?) it's kind of weird to pick a number as small as `100` for infinite fields. But then what number to pick? (`100` is what `QuickCheck` uses for `Int`).

Reply:
The characteristic isn't exactly the biggest possible number: it's the first positive integer that maps to zero (or zero if there is no such integer, in which case the field contains all of the rationals).  In a finite field the characteristic is always some prime p, and then I suppose the largest number is p-1; but that'll be equal to -1 in the field.  I don't know exactly what a typical element of a finite field would be:  0, 1, and -1 all seem a bit atypical, but in Z_2 or Z_3 those are the only elements!  100 or 101 seems reasonable enough. If it turns out to be important I suppose we could require instances of `Field` to declare a default element themselves, or at least give them the option to override some, er, default default. I was just worried about things that turn into zero because that could give rise to division by zero errors.